webapp/src/snippetBuilderInputHandler.tsx

```diff
@@ -172,39 +179,74 @@ class RangeInput extends data.Component<IRangeInputProps, {}> {
-                        <div className='left floated column snippet-slider'>
-                            <input
-                                type='range'
-                                autoFocus={autoFocus}
-                                className={'slider blocklyMockSlider'}
-                                role={'slider'}
-                                max={input.max}
-                                min={input.min}
-                                value={value}
-                                onChange={this.onChange}
-                                aria-valuemin={input.min}
-                                aria-valuemax={input.max}
-                                aria-valuenow={value}
-                                style={{
-                                    marginLeft: 0
-                                }}
-                            />
-                        </div>
-                        <div className='column slider-value snippet'>
-                            <sui.Input
-                                value={value}
-                                onChange={this.props.onChange}
-                                class='snippet slider-input'
-                            />
-                        </div>
+                    <div>
+                        <input
+                            type='range'
+                            autoFocus={autoFocus}
+                            className={'slider blocklyMockSlider'}
+                            role={'slider'}
+                            max={input.max}
+                            min={input.min}
+                            value={value}
+                            onChange={this.onChange}
+                            aria-valuemin={input.min}
+                            aria-valuemax={input.max}
+                            aria-valuenow={value}
+                            style={{
+                                marginLeft: 0
+                            }}
+                        />
+                        <sui.Input
+                            value={value}
+                            onChange={this.props.onChange}
+                            class='snippet slider-input'
+                        />
                     </div>
                 </div>
             )
         }
 
+        return null;
+    }
+}
+
+
+/**
+ * YesNo input
+ */
+
+interface IYesNoInputProps {
+    input?: pxt.SnippetQuestionInput;
+    value?: string;
+    onChange: (v: string) => void;
+}
+
+class YesNoInput extends data.Component<IYesNoInputProps, {}> {
+    constructor(props: IYesNoInputProps) {
+        super(props);
+
+        this.onChange = this.onChange.bind(this);
+    }
+
+    onChange(value: string): () => void {
+        const { onChange } = this.props;
+        return () => onChange(value);
+    }
+
+    renderCore() {
+        const { input } = this.props;
+        if (Snippet.isSnippetInputAnswerTypeYesNo(input)) {
```

Comment:
is the "input" prop only used in this check? maybe the yesno input component should just render the yes/no buttons and bind the listener, and the parent should control whether it renders?

Reply:
Yeah, I agree with you, but for the other input types the input is used more. I think the parent should have controlled the rendering on all of these but I'm not going to change paradigm now. It's wonky but it works :)